applications/GeoMechanicsApplication/custom_constitutive/incremental_linear_elastic_interface_law.h

```diff
@@ -24,6 +25,17 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) GeoIncrementalLinearElasticInterface
 public:
     using BaseType = ConstitutiveLaw;
 
+    explicit GeoIncrementalLinearElasticInterfaceLaw(std::unique_ptr<ConstitutiveLawDimension> rConstitutiveLawDimension)
+        : mConstitutiveLawDimension(std::move(rConstitutiveLawDimension))
+    {
+    }
+
+    ~GeoIncrementalLinearElasticInterfaceLaw() override = default;
+    GeoIncrementalLinearElasticInterfaceLaw(const GeoIncrementalLinearElasticInterfaceLaw&) = delete;
+    GeoIncrementalLinearElasticInterfaceLaw& operator=(const GeoIncrementalLinearElasticInterfaceLaw&) = delete;
+    GeoIncrementalLinearElasticInterfaceLaw(GeoIncrementalLinearElasticInterfaceLaw&&) = delete;
+    GeoIncrementalLinearElasticInterfaceLaw& operator=(GeoIncrementalLinearElasticInterfaceLaw&&) = delete;
```

Comment:
Since C++17, it is recommended to [make the move operations `noexcept`](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c66-make-move-operations-noexcept):
```suggestion
    GeoIncrementalLinearElasticInterfaceLaw(GeoIncrementalLinearElasticInterfaceLaw&&) noexcept = delete;
    GeoIncrementalLinearElasticInterfaceLaw& operator=(GeoIncrementalLinearElasticInterfaceLaw&&) noexcept = delete;
```
Also, is there any particular reason why we'd delete the move operations? (For the copy operations, I can see why. But for the move operations, I don't see it.)

Reply:
I've added move constructors and unit tests for them.